TST Use global_random_seed in test_huber.py

ArturoSbr opened the pull request:
I modified the file `sklearn/linear_model/tests/test_huber.py` by replacing all instances of hardcoded random seeds (ie `random_state=0`) with `global_random_seed` (in accordance to [this issue](https://github.com/scikit-learn/scikit-learn/issues/22827)).

ArturoSbr committed:
Replaced all hardcoded randon seeds with global_random_seed

ArturoSbr committed:
Black compliant file

ArturoSbr committed:
empty commit to trigger all CI jobs [all random seeds]
test_parallel

ArturoSbr commented:
I also added an empty commit to trigger all builds.

Rishab260 reviewed (changes requested):
Hi @ArturoSbr, I think your empty commit should look like this
```
test_huber [all random seeds]
make_regression_with_outliers
test_huber_equals_lr_for_high_epsilon
test_huber_max_iter
test_huber_gradient
test_huber_sample_weights
test_huber_sparse
test_huber_scaling_invariant
test_huber_and_sgd_same_results
test_huber_warm_start
test_huber_better_r2_score
test_huber_bool
```
For reference:
```
<title> [all random seeds]
<test_name_1>
<test_name_2>
...
```
from https://github.com/scikit-learn/scikit-learn/issues/22827
and https://github.com/scikit-learn/scikit-learn/pull/23388/commits/b74c8654df24ae78cc196a72169aa0e9a27cce6d

ArturoSbr committed:
Merge branch 'scikit-learn:main' into feature/global_random_seed_huber_tests

ArturoSbr committed:
Empty commit to trigger CI jobs
- test_huber [all random seeds]
- make_regression_with_outliers
- test_huber_equals_lr_for_high_epsilon
- test_huber_max_iter
- test_huber_gradient
- test_huber_sample_weights
- test_huber_sparse
- test_huber_scaling_invariant
- test_huber_and_sgd_same_results
- test_huber_warm_start
- test_huber_better_r2_score
- test_huber_bool

ArturoSbr committed:
Merge branch 'main' into feature/global_random_seed_huber_tests

ArturoSbr commented:
Hi @Rishab260! Does this PR need any additional changes?

Rishab260 commented:
> Hi @Rishab260! Does this PR need any additional changes?

Hi @ArturoSbr , I can see that this has 7 tests failing. You might want to look into why these tests have failed.
